pwa/app/lib/matchUtils.ts

```diff
@@ -17,6 +17,12 @@ const COMP_LEVEL_SHORT_STRINGS = {
   qm: 'Quals',
 };
 
+export function isValidMatchKey(key: string) {
+  return /^[1-9]\d{3}[a-z]+[0-9]*\_(?:qm|ef\d{1,2}m|qf\d{1,2}m|sf\d{1,2}m|f\dm)\d+$/.test(
```

Comment:
Can we add a couple sample unit test cases just to be sure, sometimes regex engines are different between languages